tracker_package/tracker/src/tracker/optical_flow.py

```diff
@@ -0,0 +1,81 @@
+import cv2 
+import numpy as np
+
+class VisualTrackerKLT():
+    
+    class Params:
+        def __init__(self):
+
+            self.max_corners_total = 500
+            self.max_corners_per_object = 8
+            self.mid_corners_per_object = 4
+            self.min_corners_per_object = 2
+            self.bidirectional_enable = True
+            self.bidirectional_thresh = 2.0
+            self.feature_update_rate = 10
+
+            self.detection_params = dict(maxCorners=500,
+                                        qualityLevel=0.01,
+                                        minDistance=21,
+                                        blockSize=7)
+            self.tracking_params = dict(winSize=(15, 15),
+                                       maxLevel=2,
+                                       criteria=(cv2.TERM_CRITERIA_EPS | cv2.TERM_CRITERIA_COUNT, 10, 0.03),
+                                       minEigThreshold=5e-4)
+
+    def __init__(self, params = Params()):
+        self._params = params
+        self._prev_pyr = None
+        self._kernel_d = np.ones((self._params.detection_params["minDistance"],
+                                  self._params.detection_params["minDistance"]), dtype=np.uint8)
+        self._frame_num = 0
+
+    def init(self, image, roi):
+        gray = cv2.cvtColor(image,cv2.COLOR_BGR2GRAY)
+        self._prev_pyr = gray
+        self.roi = roi
+        self.features = cv2.goodFeaturesToTrack(gray , **self._params.detection_params)
+    
+    def __call__(self, image):
+        gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+        p0 = np.squeeze(self.features)
```

Comment:
You can use a better name than "p0" for feature points array. For instance, you can use "feature_points" instead.